### PR TITLE
chore: add `make lint.fix-golint` to address auto fixable lint issues

### PR DESCRIPTION
**Pull request by sanposhiho**
**What type of PR is this?**

**What this PR does / why we need it**:

This PR adds `make lint.fix-golint` to address auto-fixable lint issues.
`gci`, which is the most annoying one for me, has to be handled separately since it doesn't support auto-fix in golangci-lint, somehow.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Comment by zirain** [comment]
`lint.fix-golint` should be the name?

File: tools/src/gci/go.mod [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module local
+
+go 1.22.4
```
Comment (zirain) [inline tools/src/gci/go.mod:3]: plesae use 1.22.5 for CVE

**Comment by sanposhiho** [comment]
@zirain Addressed your points.

File: tools/make/lint.mk [hunk 2]
```diff
@@ -68,6 +68,13 @@ lint.shellcheck: $(tools/shellcheck)
 	@$(LOG_TARGET)
 	$(tools/shellcheck) tools/hack/*.sh
 
+.PHONY: lint.fix-golint
+lint-deps: $(tools/gci)
+lint.fix: ## Run all linter of code sources and fix the issues.
```
Comment (zirain) [inline tools/make/lint.mk:73]:
```suggestion
lint.fix-golint: ## Run all linter of code sources and fix the issues.
```

Comment (sanposhiho) [reply tools/make/lint.mk]: 🤦‍♂️

File: tools/make/lint.mk [hunk 3]
```diff
@@ -68,6 +68,13 @@ lint.shellcheck: $(tools/shellcheck)
 	@$(LOG_TARGET)
 	$(tools/shellcheck) tools/hack/*.sh
 
+.PHONY: lint.fix-golint
+lint-deps: $(tools/gci)
+lint.fix-golint: ## Run all linter of code sources and fix the issues.
+	@$(LOG_TARGET)
+	$(MAKE) lint.golint GOLANGCI_LINT_FLAGS="--fix"
+	find . -name "*.go" | xargs $(tools/gci) write --skip-generated -s Standard -s Default -s "Prefix(github.com/envoyproxy/gateway)"
```
Comment (zirain) [inline tools/make/lint.mk:76]: I thinks the `Prefix` should be consistent with L74 in `tools/linter/golangci-lint/.golangci.yml`

**Comment by sanposhiho** [comment]
`gofumpt`'s autofix should be supported within golangci-lint, unlike gci.

**Review by guydc — APPROVED** [review]
LGTM

**Pull request merged**